modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/metal/js/components/Calculator/Calculator.es.js

```diff
@@ -0,0 +1,231 @@
+		).value([]),
+
+		strings: {
+			value: {
+				addField: Liferay.Language.get('add-field')
+			}
+		}
+	}
+
+	attached() {
+		this._resetExpression();
+	}
+
+	willReceiveState(changes) {
+		let {expression, expressionArray} = this;
+
+		if (changes.hasOwnProperty('resultSelected')) {
+			this._resetExpression();
+		}
+		else if (changes.hasOwnProperty('options')) {
+			if (changes.expression) {
+				expression = changes.expression.prevVal;
+			}
+			else {
+				expression = '';
+			}
+
+			expressionArray = changes.expressionArray.prevVal;
+
+			this.setState(
+				{
+					expression,
+					expressionArray
+				}
+			);
+		}
+	}
+
+	_resetExpression() {
+		let {expression, expressionArray} = this;
+
+		expression = '';
+		expressionArray = [];
+
+		this.setState(
+			{
+				expression,
+				expressionArray
+			}
+		);
+	}
+
+	_formatExpression(expressionArray, expression) {
+		const regexRepeatableArithSigns = /[\+\-\*/]{2,}/;
+		const regexRepeatableDotSign = /[\.]{2,}/;
+
+		if (expression.match(regexRepeatableArithSigns) || expression.match(regexRepeatableDotSign)) {
+			const lastChar = expressionArray.pop();
+
+			expressionArray.pop();
+			expressionArray.push(lastChar);
+		}
+
+		return expressionArray;
+	}
+
+	_setFieldsRepeatable() {
+		const {options} = this;
+
+		let {optionsRepeatable} = this;
+
+		optionsRepeatable = options.filter(
+			option => {
+				return option.repeatable == true;
+			}
+		);
+
+		this.setState(
+			{
+				optionsRepeatable
+			}
+		);
+	}
+
+	_handleItemSelection(event) {
+		let {disableCalculatorField, expressionArray} = this;
+		let calculatorOperationFieldSelected = false;
+		let calculatorSymbol = '';
+		let dropdownItemName = '';
+
+		const keyWasClicked = event.target.dataset;
+
+		const dropdownItemWasSelected = event.data;
+
+		if (dropdownItemWasSelected) {
+			dropdownItemName = event.data.item.fieldName;
+
+			if (!dropdownItemName) {
+				calculatorOperationFieldSelected = true;
+				dropdownItemName = event.data.item.value;
+			}
+		}
+		else if (keyWasClicked) {
+			calculatorSymbol = event.target.dataset.calculatorKey;
+		}
+
+		if (expressionArray) {
+
+			if (dropdownItemWasSelected) {
+				if (calculatorOperationFieldSelected) {
+					this._setFieldsRepeatable();
+
+					disableCalculatorField = true;
+
+					expressionArray.push(`${dropdownItemName}(`);
+				}
+				else if (disableCalculatorField) {
+					expressionArray.push(`${dropdownItemName})`);
+
+					disableCalculatorField = false;
+				}
+				else {
+					expressionArray.push(dropdownItemName);
+				}
+			}
+			else if (calculatorSymbol == 'backspace') {
+				expressionArray.pop();
+
+				const lastElement = expressionArray[expressionArray.length - 1];
+
+				if (lastElement && lastElement.indexOf('(') > 0) {
+					disableCalculatorField = true;
+				}
+				else {
+					disableCalculatorField = false;
+				}
+			}
+			else if (calculatorSymbol) {
+				expressionArray.push(calculatorSymbol);
+			}
+		}
+
+		if (expressionArray) {
```

Comment:
Since you put that line `expressionArray = this._formatExpression(expressionArray, expressionArray.join(''));` inside the if declaration, you can remove this code block.